tests/ixia/test_pfc_pause_lossless.py

```diff
@@ -0,0 +1,289 @@
+    if send_pause_frame:
+        pfc_traffic = create_pause_traffic(session=session,
+                                           name='PFC Pause Storm',
+                                           source=rx_port,
+                                           pkt_per_sec=1.1/pause_dur_per_pkt,
+                                           start_delay=0,
+                                           global_pause=False,
+                                           pause_prio_list=test_prio_list)
+
+    start_traffic(session)
+
+    # Wait for test and background traffic to finish.
+    time.sleep(exp_dur + start_delay)
+
+    # Capture traffic statistics.
+    flow_statistics = get_traffic_statistics(session)
+    logger.info(flow_statistics)
+
+    for row_number, flow_stat in enumerate(flow_statistics.Rows):
+        tx_frames = int(flow_stat['Tx Frames'])
+        rx_frames = int(flow_stat['Rx Frames'])
+
+        if 'Test' in flow_stat['Traffic Item']:
+            if test_traffic_pause_expected:
+                pytest_assert(tx_frames > 0 and rx_frames == 0,
+                    "Test traffic should be fully paused")
+            else:
+                pytest_assert(tx_frames > 0 and tx_frames == rx_frames,
+                    "Test traffic should not be impacted")
+
+        elif 'PFC' in flow_stat['Traffic Item']:
+            pytest_assert(tx_frames > 0 and rx_frames == 0,
+                "PFC packets should be dropped")
+        else:
+            pytest_assert(tx_frames > 0 and tx_frames == rx_frames,
+                "Background traffic should not be impacted")
+
+    stop_traffic(session)
+
+
+def test_pfc_pause_lossless(testbed, conn_graph_facts, lossless_prio_dscp_map,
+                            duthost, ixia_dev, ixia_api_server_session,
+                            fanout_graph_facts):
+    """
+    RDMA PFC - Pauses on lossless priorities.
+    1. On SONiC DUT enable PFC on any two priorities say, m and n.
+    2. Disable the PFC watchdog on the SONiC DUT.
+    3. On the Ixia Tx port create two flows - a) 'Test Data Traffic' and
+       b) 'Background Data traffic'.
+    4. The flow 'Test Data Traffic' can assume only one of the priority 
+       values - either m or n (or both).
+    5. The flow 'Background Data Traffic' can assume any priority values 
+       which is not in 'Test Data Traffic' (including m or n). Note that:
+       a. Background data traffic can assume priority n if 'Test Data Traffic'
+          has the TC value m.
+       b. Background data traffic can assume priority m if 'Test Data Traffic'
+          has the TC value n.
+       c. Neither m or n when 'Test Data Traffic' has both the priorities.   
+    6. Start 'Test Data Traffic' and 'Background Data Traffic'.
+    7. From Rx port send pause frames on priorities either m or n. Such that
+       priority of 'Test Data Traffic' at Tx end == Pause Priority at Rx end.
+    8. You may repeat the steps 6 and 7 several times.
```

Comment:
Repeat the steps 6 and 7 to test each lossless priority.

Reply:
This comment is implemented